src/test/cypress/cypress/integration/GeoMap.spec.js

```diff
@@ -0,0 +1,186 @@
+/* Copyright (c) 2023, Alliander (http://www.alliander.com)
+ * See AUTHORS.txt
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of the OperatorFabric project.
+ */
+
+import {OpfabGeneralCommands} from "../support/opfabGeneralCommands";
+import {ScriptCommands} from "../support/scriptCommands";
+
+describe('GeoMap tests', function () {
+
+    const opfab = new OpfabGeneralCommands();
+    const script = new ScriptCommands();
+
+    beforeEach('Reset UI configuration file ', function () {
+        script.resetUIConfigurationFiles();
+        script.deleteAllCards();
+    })
+
+    before('Set configuration', function () {
+        script.loadTestConf();
+        script.deleteAllCards();
+    })
+
+    describe('Check enable and disable of the geomap', function () {
+        it('geomap disabled -> geomap should not be visible in the feed', () => {
+            script.setPropertyInConf('feed.geomap.enableMap', 'web-ui', false);
+            opfab.loginWithUser('operator1_fr');
+            cy.get('of-map').should('not.be.visible');
+        });
+
+        it('geomap enabled -> geomap should be visible in the feed', () => {
+            script.setPropertyInConf('feed.geomap.enableMap', 'web-ui', true);
+            opfab.loginWithUser('operator1_fr');
+            cy.get('of-map').should('be.visible');
+        });
+    });
+
+    describe('Check enable and disable of the graph on the geomap', function () {
+        it('enableGraph disabled -> graph should not be visible on the geomap in the feed', () => {
+            script.setPropertyInConf('feed.geomap.enableMap','web-ui', true);
+            script.setPropertyInConf('feed.geomap.enableGraph','web-ui', false);
+            opfab.loginWithUser('operator1_fr');
+            script.sendCard('cypress/geoMap/point_card1.json');
+            cy.get('#mapGraph').should('not.exist');
+        });
+
+        it('enableGraph enabled -> graph should be visible on the geomap in the feed', () => {
+            script.setPropertyInConf('feed.geomap.enableMap','web-ui', true);
+            script.setPropertyInConf('feed.geomap.enableGraph','web-ui', true);
+            opfab.loginWithUser('operator1_fr');
+            script.sendCard('cypress/geoMap/point_card1.json');
+            cy.get('#mapGraph').should('exist');
+        });
+    });
+
+    describe('Check if points are shown on the geomap', function () {
+        it('Card with point data is pointed on the geomap', () => {
+            script.setPropertyInConf('feed.geomap.enableMap', 'web-ui', true);
+            opfab.loginWithUser('operator1_fr');
+            script.sendCard('cypress/geoMap/point_card1.json');
+            cy.wait(3000);
```

Comment:
Is deze wait waarde gekoppeld aan een specifieke wait waarde in de code of wordt het gebruikt om te wachten op een response van een backend request? In het laatste geval kan het soms zijn dat deze waarde niet voldoende is (wnr een server traag is) en je een error hier krijgt. Is het niet mogelijk om .then te gebruiken? https://docs.cypress.io/api/commands/then

Reply:
@mobilabgit De wait is ingebouwd vanwege dat de geomap inzoomt bij het openen van de card feed en dit kost een paar seconden. Als de code meteen doorgaat dan lopen er twee zooms door elkaar en wordt het juiste punt niet altijd weergeven. Met een wachttijd loopt dit wel juist.